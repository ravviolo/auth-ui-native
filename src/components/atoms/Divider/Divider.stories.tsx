import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import { StoryScreen } from 'storybook/StoryScreen';

import { Divider } from './Divider';

storiesOf('Divider', module)
  .addDecorator((getStory) => <StoryScreen>{getStory()}</StoryScreen>)
  .add('With Label', () => (
    <Divider label={text('label', 'Label')} testID={text('testID', 'divider-test-id')} />
  ))
  .add('No Label', () => <Divider testID={text('testID', 'divider-test-id')} />);
