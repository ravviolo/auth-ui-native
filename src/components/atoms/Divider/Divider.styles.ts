import { StyleSheet } from 'react-native';
import { colors } from 'theme';

export const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  label: {
    backgroundColor: colors.background,
    borderColor: colors.textLight,
    borderRadius: 5,
    borderWidth: 1,
    color: colors.textLight,
    padding: 4,
    position: 'absolute',
    textAlign: 'center',
    textTransform: 'uppercase',
    zIndex: 2,
  },
  line: {
    alignSelf: 'stretch',
    backgroundColor: colors.textLight,
    height: 1,
  },
});
