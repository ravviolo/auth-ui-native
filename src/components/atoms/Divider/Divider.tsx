import { StyleSheet, Text, View } from 'react-native';
import { colors } from 'theme';

export interface Props {
  testID: string;
  label?: string;
}

export const Divider = ({ label, testID }: Props) => {
  return (
    <View style={styles.container} testID={testID}>
      <View style={styles.line} />
      <Text style={styles.label}>{label}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  label: {
    backgroundColor: colors.background,
    borderColor: colors.textLight,
    borderRadius: 5,
    borderWidth: 1,
    color: colors.textLight,
    padding: 4,
    position: 'absolute',
    textAlign: 'center',
    zIndex: 2,
  },
  line: {
    alignSelf: 'stretch',
    backgroundColor: colors.textLight,
    height: 1,
  },
});
