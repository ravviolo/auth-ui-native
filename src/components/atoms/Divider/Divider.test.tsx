import { render, screen, within } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { Divider } from './Divider';

import type { Props } from './Divider';

describe('Divider', () => {
  const props: Props = {
    testID: 'divider-test-id',
  };

  it('should render correctly without label', () => {
    render(<Divider {...props} />);

    const divider = screen.getByTestId('divider-test-id');

    expect(divider).not.toBeEmpty();

    const labelText = within(divider).queryByText('Label');

    expect(labelText).toBeNull();
  });

  it('should render correctly with label', () => {
    render(<Divider {...props} label="OR" />);

    const divider = screen.getByTestId('divider-test-id');

    expect(divider).not.toBeEmpty();

    const labelText = screen.getByText('OR');

    expect(divider).toContainElement(labelText);
  });

  it('should match snapshot without label', () => {
    const tree = testRenderer.create(<Divider {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with label', () => {
    const tree = testRenderer.create(<Divider {...props} label="OR" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
