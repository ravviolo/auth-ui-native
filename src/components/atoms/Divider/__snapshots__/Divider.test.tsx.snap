// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider should match snapshot with label 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
  testID="divider-test-id"
>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#9E9E9E",
        "height": 1,
      }
    }
  />
  <Text
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#9E9E9E",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#9E9E9E",
        "padding": 4,
        "position": "absolute",
        "textAlign": "center",
        "textTransform": "uppercase",
        "zIndex": 2,
      }
    }
  >
    OR
  </Text>
</View>
`;

exports[`Divider should match snapshot without label 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
  testID="divider-test-id"
>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#9E9E9E",
        "height": 1,
      }
    }
  />
</View>
`;
